trieste/space.py

```diff
@@ -329,3 +343,121 @@ def __mul__(self, other: Box) -> Box:
 
     def __deepcopy__(self, memo: dict[int, object]) -> Box:
         return self
+
+
+class DecomposableSearchSpace(SearchSpace):
+    r"""
+    Decomposable :class:`SearchSpace` consisting of separate decision and
+    context :class:`SearchSpace`s. This class provides functionality for using
+    either the resulting combined search space or each individual space.
+
+    Note that this class assumes individual points are represented with their context
```

Comment:
This seems like potential problem down the line, it will make things more difficult for any downstream function that needs to differentiate between them (models, acquisiton functions etc). Why not having a `NamedTuple` of tensors? Then those functions that are able or need to use context will use the additional attribute in the tuple.

Reply:
I get that, I'm just thinking whether we might want to move away from that - I didn't spell it out above, but if we start using `NamedTuple` here we would probably need to use them for other search spaces, but I think that might not be too much of a change

perhaps its too hard to say what would be more useful without seeing models/acquisitions examples that would handle both context and decision spaces